### PR TITLE
SLCORE-157 Update to SQ Plugin API 6.7

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java [hunk 1]
```diff
@@ -78,10 +77,6 @@ public InputDir inputDir(String relativePath) {
     return null;
   }
 
-  public InputFile inputFile(Path path) {
-    return inputFileCache.get(path);
-  }
-
   public InputDir inputDir(Path path) {
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java]: MINOR Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java [hunk 2]
```diff
@@ -19,34 +19,33 @@
...
   private final SetMultimap<String, InputFile> filesByExtensionCache = LinkedHashMultimap.create();
   private final SortedSet<String> languages = new TreeSet<>();
 
   @Override
   public Iterable<InputFile> inputFiles() {
-    return inputFileCache.values();
+    return inputFileCache;
   }
 
   public Iterable<InputDir> allDirs() {
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java]: MINOR Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java [hunk 3]
```diff
@@ -19,34 +19,33 @@
...
 
-import com.google.common.collect.LinkedHashMultimap;
-import com.google.common.collect.SetMultimap;
-
 @SonarLintSide
 public class InputPathCache extends DefaultFileSystem.Cache {
 
-  private final Map<Path, InputFile> inputFileCache = new LinkedHashMap<>();
+  private final Set<InputFile> inputFileCache = new LinkedHashSet<>();
   private final Map<Path, InputDir> inputDirCache = new LinkedHashMap<>();
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java]: MINOR Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java [hunk 4]
```diff
@@ -105,9 +106,9 @@ public boolean matchFilePattern(String antPattern) {
 
   private final Project project;
 
-  public LtsApiSensorContext(Project project, InputModule inputModule, Settings settings, FileSystem fs, ActiveRules activeRules, SensorStorage sensorStorage,
+  public LtsApiSensorContext(Project project, InputModule inputModule, Settings settings, Configuration config, FileSystem fs, ActiveRules activeRules, SensorStorage sensorStorage,
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java]: MINOR Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/AbsolutePathPredicate.java [hunk 5]
```diff
@@ -42,14 +35,4 @@ public boolean apply(InputFile f) {
     return path.equals(f.absolutePath());
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/AbsolutePathPredicate.java]: MINOR Remove this use of "absolutePath"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java [hunk 6]
```diff
@@ -59,11 +60,13 @@
   private final InputModule module;
   private final SonarRuntime sqRuntime;
   private final ProgressWrapper progress;
+  private final Configuration config;
 
-  public DefaultSensorContext(InputModule module, Settings settings, FileSystem fs, ActiveRules activeRules, SensorStorage sensorStorage,
+  public DefaultSensorContext(InputModule module, Settings settings, Configuration config, FileSystem fs, ActiveRules activeRules, SensorStorage sensorStorage,
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java]: MAJOR Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)
MINOR Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 19 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 18 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [DefaultSensorContext.java#L56](https://github.com/SonarSource/sonarlint-core/blob/287863e9ee8f0177a67380af27fe489e98dcf194/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java#L56): Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [DefaultSensorContext.java#L78](https://github.com/SonarSource/sonarlint-core/blob/287863e9ee8f0177a67380af27fe489e98dcf194/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java#L78): Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [LtsApiSensorContext.java#L63](https://github.com/SonarSource/sonarlint-core/blob/287863e9ee8f0177a67380af27fe489e98dcf194/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java#L63): Remove this use of "absolutePath"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [LtsApiSensorContext.java#L225](https://github.com/SonarSource/sonarlint-core/blob/287863e9ee8f0177a67380af27fe489e98dcf194/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java#L225): Remove this use of "absolutePath"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [LtsApiSensorContext.java#L230](https://github.com/SonarSource/sonarlint-core/blob/287863e9ee8f0177a67380af27fe489e98dcf194/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java#L230): Remove this use of "path"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [InputPathCache.java#L66](https://github.com/SonarSource/sonarlint-core/blob/287863e9ee8f0177a67380af27fe489e98dcf194/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java#L66): Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [InputPathCache.java#L76](https://github.com/SonarSource/sonarlint-core/blob/287863e9ee8f0177a67380af27fe489e98dcf194/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java#L76): Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [SonarLintFileSystem.java#L47](https://github.com/SonarSource/sonarlint-core/blob/287863e9ee8f0177a67380af27fe489e98dcf194/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/SonarLintFileSystem.java#L47): Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [SonarLintInputFile.java#L53](https://github.com/SonarSource/sonarlint-core/blob/287863e9ee8f0177a67380af27fe489e98dcf194/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/SonarLintInputFile.java#L53): Don't override a deprecated method or explicitly mark it as "@Deprecated". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [DefaultServer.java#L34](https://github.com/SonarSource/sonarlint-core/blob/287863e9ee8f0177a67380af27fe489e98dcf194/core/src/main/java/org/sonarsource/sonarlint/core/container/connected/DefaultServer.java#L34): Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 19 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 18 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [DefaultSensorContext.java#L56](https://github.com/SonarSource/sonarlint-core/blob/cc3ef8e0ca6ef237f3e0576b5b589cc1ff5b2525/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java#L56): Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [DefaultSensorContext.java#L78](https://github.com/SonarSource/sonarlint-core/blob/cc3ef8e0ca6ef237f3e0576b5b589cc1ff5b2525/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java#L78): Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [LtsApiSensorContext.java#L63](https://github.com/SonarSource/sonarlint-core/blob/cc3ef8e0ca6ef237f3e0576b5b589cc1ff5b2525/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java#L63): Remove this use of "absolutePath"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [LtsApiSensorContext.java#L225](https://github.com/SonarSource/sonarlint-core/blob/cc3ef8e0ca6ef237f3e0576b5b589cc1ff5b2525/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java#L225): Remove this use of "absolutePath"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [LtsApiSensorContext.java#L230](https://github.com/SonarSource/sonarlint-core/blob/cc3ef8e0ca6ef237f3e0576b5b589cc1ff5b2525/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java#L230): Remove this use of "path"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [InputPathCache.java#L66](https://github.com/SonarSource/sonarlint-core/blob/cc3ef8e0ca6ef237f3e0576b5b589cc1ff5b2525/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java#L66): Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [InputPathCache.java#L76](https://github.com/SonarSource/sonarlint-core/blob/cc3ef8e0ca6ef237f3e0576b5b589cc1ff5b2525/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java#L76): Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [SonarLintFileSystem.java#L47](https://github.com/SonarSource/sonarlint-core/blob/cc3ef8e0ca6ef237f3e0576b5b589cc1ff5b2525/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/SonarLintFileSystem.java#L47): Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [SonarLintInputFile.java#L53](https://github.com/SonarSource/sonarlint-core/blob/cc3ef8e0ca6ef237f3e0576b5b589cc1ff5b2525/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/SonarLintInputFile.java#L53): Don't override a deprecated method or explicitly mark it as "@Deprecated". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [DefaultServer.java#L34](https://github.com/SonarSource/sonarlint-core/blob/cc3ef8e0ca6ef237f3e0576b5b589cc1ff5b2525/core/src/main/java/org/sonarsource/sonarlint/core/container/connected/DefaultServer.java#L34): Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java [hunk 7]
```diff
@@ -76,6 +79,11 @@ public Settings settings() {
     return settings;
   }
 
+  @Override
+  public Configuration config() {
+    return null;
```
Comment (janos-ss) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java]: Shouldn't it be `return config` ?

File: core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/SonarLintPathPattern.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+public class SonarLintPathPattern {
+
+  private static final Logger LOG = Loggers.get(SonarLintPathPattern.class);
+
+  final WildcardPattern pattern;
+
+  public SonarLintPathPattern(String pattern) {
+    if (pattern.startsWith("file:")) {
+      LOG.warn("Unsupported path pattern: " + pattern);
+      pattern = pattern.substring("file:".length());
```
Comment (janos-ss) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/SonarLintPathPattern.java]: If the pattern starts with `file:`, it's likely to have multiple `/` after `file:`, no? In which case, wouldn't it be good to strip those here, that is `pattern = pattern.replaceAll("^file:/*", "")`

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java (hunk 7 above)
Comment (dbmeneses) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java]: null?

**Comment by dbmeneses** [comment]
With the cleanup of old API, the plugin minimal supported versions might need to be changed (ITs will tell)

**Comment by henryju** [comment]
Thanks for spotting the null. I quickly checked, and seems like minimal plugins versions are already 5.6+.

**Comment by SonarTech** [comment]
SonarQube analysis reported 19 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 18 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [DefaultSensorContext.java#L56](https://github.com/SonarSource/sonarlint-core/blob/3db0176387a030c04891a65eb3683eeb02dd6590/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java#L56): Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [DefaultSensorContext.java#L78](https://github.com/SonarSource/sonarlint-core/blob/3db0176387a030c04891a65eb3683eeb02dd6590/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java#L78): Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [LtsApiSensorContext.java#L63](https://github.com/SonarSource/sonarlint-core/blob/3db0176387a030c04891a65eb3683eeb02dd6590/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java#L63): Remove this use of "absolutePath"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [LtsApiSensorContext.java#L225](https://github.com/SonarSource/sonarlint-core/blob/3db0176387a030c04891a65eb3683eeb02dd6590/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java#L225): Remove this use of "absolutePath"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [LtsApiSensorContext.java#L230](https://github.com/SonarSource/sonarlint-core/blob/3db0176387a030c04891a65eb3683eeb02dd6590/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java#L230): Remove this use of "path"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [InputPathCache.java#L66](https://github.com/SonarSource/sonarlint-core/blob/3db0176387a030c04891a65eb3683eeb02dd6590/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java#L66): Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [InputPathCache.java#L76](https://github.com/SonarSource/sonarlint-core/blob/3db0176387a030c04891a65eb3683eeb02dd6590/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java#L76): Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [SonarLintFileSystem.java#L47](https://github.com/SonarSource/sonarlint-core/blob/3db0176387a030c04891a65eb3683eeb02dd6590/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/SonarLintFileSystem.java#L47): Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [SonarLintInputFile.java#L53](https://github.com/SonarSource/sonarlint-core/blob/3db0176387a030c04891a65eb3683eeb02dd6590/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/SonarLintInputFile.java#L53): Don't override a deprecated method or explicitly mark it as "@Deprecated". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [DefaultServer.java#L34](https://github.com/SonarSource/sonarlint-core/blob/3db0176387a030c04891a65eb3683eeb02dd6590/core/src/main/java/org/sonarsource/sonarlint/core/container/connected/DefaultServer.java#L34): Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 19 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 18 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [DefaultSensorContext.java#L56](https://github.com/SonarSource/sonarlint-core/blob/e40282d7629ed86db0e10930e10c5bdd606fdf56/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java#L56): Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [DefaultSensorContext.java#L78](https://github.com/SonarSource/sonarlint-core/blob/e40282d7629ed86db0e10930e10c5bdd606fdf56/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/DefaultSensorContext.java#L78): Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [LtsApiSensorContext.java#L63](https://github.com/SonarSource/sonarlint-core/blob/e40282d7629ed86db0e10930e10c5bdd606fdf56/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java#L63): Remove this use of "absolutePath"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [LtsApiSensorContext.java#L225](https://github.com/SonarSource/sonarlint-core/blob/e40282d7629ed86db0e10930e10c5bdd606fdf56/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java#L225): Remove this use of "absolutePath"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [LtsApiSensorContext.java#L230](https://github.com/SonarSource/sonarlint-core/blob/e40282d7629ed86db0e10930e10c5bdd606fdf56/core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java#L230): Remove this use of "path"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [InputPathCache.java#L66](https://github.com/SonarSource/sonarlint-core/blob/e40282d7629ed86db0e10930e10c5bdd606fdf56/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java#L66): Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [InputPathCache.java#L76](https://github.com/SonarSource/sonarlint-core/blob/e40282d7629ed86db0e10930e10c5bdd606fdf56/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/InputPathCache.java#L76): Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [SonarLintFileSystem.java#L47](https://github.com/SonarSource/sonarlint-core/blob/e40282d7629ed86db0e10930e10c5bdd606fdf56/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/SonarLintFileSystem.java#L47): Remove this use of "InputDir"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [SonarLintInputFile.java#L53](https://github.com/SonarSource/sonarlint-core/blob/e40282d7629ed86db0e10930e10c5bdd606fdf56/core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/SonarLintInputFile.java#L53): Don't override a deprecated method or explicitly mark it as "@Deprecated". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [DefaultServer.java#L34](https://github.com/SonarSource/sonarlint-core/blob/e40282d7629ed86db0e10930e10c5bdd606fdf56/core/src/main/java/org/sonarsource/sonarlint/core/container/connected/DefaultServer.java#L34): Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**